### ISSUE TITLE
VP8 video frame getting blured

### ISSUE THREAD

**Issue by pnkjvvdy** [medooze/media-server#12]
I am working with Video conference server application. I join conference from Linphone 3.7.0 client to my server using VP8 codec with HD720p(1280x720) resolution. FPS is 30 and bitrate is 1500. Call established successfully and at client side I am getting my own video. After sometime I send video mute request to my server so server sends me predefined image. Internally server receives video frames from client and decode it and decoded frames over written into one buffer. Now after video un-mute server pickup latest decoded frames from buffer and passed it to VP8 encoder and this encoded frame sent to the client.

But problem is that after video unmute I found blurred displays at client side.

My code design is as following Receiving RTP packets from client -> decode video frame into YUV420 -> Resizing raw data if required and mixed into layout -> Encode YUV420 raw data into VP8 -> Sending Encoded frame to client using RTP

I use vpx-1.4.0 library for VP8 encoding and decoding process.

**Comment by murillo128** [comment]
But are you using my server?

**Comment by pnkjvvdy** [comment]
Thanks for reply and First sorry for my poor English

Yes I am using medooze/mcu server. After long hardworking and tanking number of observation we found following things
On video mute Receiving thread stop placing participant video frame into PipeVideoOutput. So mixer always mixed last frame stored into PipeVideoOutput. In mixer thread we place video block image into participant slot and this mixed frame sent to all participant.
In case of VP8 codec decoder has reference frame and it decodes new frame by using this reference frame.
After unmute video we'll suddenly send mixed image with participant video so client(Linphone 3.7.0) does not able to decode properly because it has reference frame which has totally different frame . So in this case MCU server has to send key frame to video participant who negotiated with VP8 codec.

I had done this but I had to track that after changing video mute state, Participant's video frame must received by mcu, this frame must be provided to PipeVideoOutput, Mixer must get this frame from PipeVideoOutput and mixed into Mosaic, this mosaic must provided to PipeVideoInput and Sending thread must get this mixed frame from PipeVideoInput. After that I can send this frame as a Key frame to all participant.

Is there any better way?
